### DISCUSSION TITLE
One-to-many Table Relationship Data in a Formula

### DISCUSSION THREAD

**Discussion by APBowls** [Budibase/budibase#9300] [Help] [answered]
Hello, 

I am in the process of creating a tool to track production and generate incentive reports using internal tables. I have a table dedicated to the product SKU #, with point values assigned to each product. 
Then through an input page, employees will input their production with the SKU and quantity produced, adding a row to a table called Bi-Weekly Production.

The Points column is a formula, shown.
![Bi-Weekly Production](https://user-images.githubusercontent.com/122388679/211640591-73701140-5331-4fa4-871a-060a209a89c3.PNG)
![Bi-Weekly Production formula](https://user-images.githubusercontent.com/122388679/211641903-fe164d97-8a1b-4da2-9c41-d76766146c1e.PNG)

This all works as expected. Now to the real question. I have another table, Pay Period. This is sorted by employee, and brings in all of the points produced through a relationship to the Bi-Weekly Production table. Is it possible to add these values together in each row using a formula in the Total Points column?

![pay period](https://user-images.githubusercontent.com/122388679/211645118-6764252b-ce17-455a-91dc-8f3d13c16ba7.PNG)

I have searched the docs, and attempted to make this work but to no avail. Any help, or alternative solutions are greatly appreciated.

**Comment by APBowls** [comment]
@melohagan, might you have any insight on this?

**Comment by melohagan** [comment, marked as answer]
Hey @APBowls 

This should be possible. This is a similar example here: https://docs.budibase.com/docs/formula#tutorial-test-scores-using-sum

A few things to watch out for:

1. Make sure 'Points' is number type in the Bi-Weekly Production table -> This means using JavaScript to return the points value as Handlebars always return strings
2. If the points formula in the Bi-Weekly table is static, then the formula column in the Pay Period table may need to be Dynamic for it to be able to reference another formula. (Formulas can only go one level of relationships deep I believe)
3. Use the aggregate sum for TotalPoints, something like: `{{ sum (pluck Points 'Points') }}` . I don't think it should matter, but you may not want to call the relationship column Bi-Weekly Production (the table it relates to) instead of Points.

**Reply by APBowls** [reply]
Thank you! This was what I needed. Changing the 'Point's formula to JavaScript to return a number type was likely the issue I was running into.

**Reply by APBowls** [reply]
Now I have another question. Can I reference the aggregate sum formula for TotalPoints inside a different formula in another column? Or do I need to do all of my arithmetic inside one formula?

**Reply by melohagan** [reply]
You may need to do it all in one formula. I think when you start to chain more than two formulas (a static formula referenced by a dynamic formula) it stops working.
